File: wxrtc/src/main/java/com/wx/rtc/WXRTCListener.java
package com.wx.rtc;

import com.wx.rtc.bean.ResultData;

import java.io.File;

public interface WXRTCListener {
    default void onError(int errCode, String errMsg){};
    default void onLogin(){};
    default void onLogout(int reason){};
    default void onEnterRoom(){};
    default void onExitRoom(int reason){};
    default void onRemoteUserEnterRoom(String userId){};
    default void onRemoteUserLeaveRoom(String userId, int reason){};
    default void onRecvRoomMsg(String userId, String cmd, String message){};
    default void onResult(ResultData resultData){};

    default void onRecordStart(String fileName){};
    default void onRecordEnd(String fileName){};
    default void onScreenShot(File file){};
}

File: wxrtc/src/main/java/com/wx/rtc/WXRTCListener.kt
package com.wx.rtc

interface WXRTCListener {
    fun onError(errCode: Int, errMsg: String) {}
    fun onLogin() {}
    fun onLogout(reason: Int) {}
    fun onEnterRoom() {}
    fun onExitRoom(reason: Int) {}
    fun onRemoteUserEnterRoom(userId: String) {}
    fun onRemoteUserLeaveRoom(userId: String, reason: Int) {}
    fun onUserVideoAvailable(userId: String, available: Boolean) {}
    fun onUserAudioAvailable(userId: String, available: Boolean) {}
    fun onRecvRoomMsg(userId: String, cmd: String, message: String) {}
    fun onProcessResult(processData: WXRTCDef.ProcessData) {}
    fun onRecordStart(fileName: String) {}
    fun onRecordEnd(fileName: String) {}
}
